frontend/src/App.js

```diff
@@ -9,26 +9,59 @@ import Landing from "pages/landing/Landing";
 import ProductOverview from "pages/product-overview/ProductOverview";
 import Search from "pages/search/Search";
 import "./App.css";
+import Login from "pages/login/Login";
+import {UserContext} from "context/UserContext";
+import Register from "pages/register/Register";
+import GuestRoute from "components/protected/GuestRoute";
+import './api/AxiosInterceptor';
 
 function App() {
+  const [user, setUser] = useState(JSON.parse(localStorage.getItem("user")));
+  const value = useMemo(() => ({user, setUser}), [user, setUser]);
+
+  useEffect(() => {
+    setUser(JSON.parse(localStorage.getItem("user")));
+  }, []);
```

Comment:
Is this necessary? The initial state is already set to the value in local storage.